src/symbol/mod.rs

```diff
@@ -1250,6 +1254,92 @@ impl<'t> TryFromCtx<'t, SymbolKind> for BlockSymbol<'t> {
     }
 }
 
+/// Thunk adjustor
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct ThunkAdjustor<'t> {
+    delta: u16,
+    target: RawString<'t>,
+}
+
+/// A thunk kind
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ThunkKind<'t> {
+    /// Standard thunk
+    NoType,
+    /// "this" adjustor thunk with delta and target
+    Adjustor(ThunkAdjustor<'t>),
+    /// Virtual call thunk with table entry
+    VCall(u16),
+    /// pcode thunk
+    PCode,
+    /// thunk which loads the address to jump to via unknown means...
+    Load,
+    /// Unknown
+    Unknown,
+}
+
+/// A thunk symbol.
+///
+/// Symbol kind `S_THUNK32`, or `S_THUNK32_ST`.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct ThunkSymbol<'t> {
+    /// The parent scope that this thunk is nested in.
+    pub parent: Option<SymbolIndex>,
+    /// The end symbol of this thunk.
+    pub end: SymbolIndex,
+    /// The next symbol.
+    pub next: Option<SymbolIndex>,
+    /// Code offset of the start of this label.
+    pub offset: PdbInternalSectionOffset,
+    /// The length of the thunk.
+    pub len: u16,
+    /// The kind of the thunk.
+    pub kind: ThunkKind<'t>,
+    /// The thunk name.
+    pub name: RawString<'t>,
+}
+
+impl<'t> TryFromCtx<'t, SymbolKind> for ThunkSymbol<'t> {
+    type Error = Error;
+    type Size = usize;
+
+    fn try_from_ctx(this: &'t [u8], kind: SymbolKind) -> Result<(Self, Self::Size)> {
+        let mut buf = ParseBuffer::from(this);
+
+        let parent = parse_optional_index(&mut buf)?;
+        let end = buf.parse()?;
+        let next = parse_optional_index(&mut buf)?;
+        let offset = buf.parse()?;
+        let len = buf.parse()?;
+        let ord = buf.parse::<u8>()?;
+        let name = parse_symbol_name(&mut buf, kind)?;
+
+        let kind = match ord {
```

Comment:
Would it be possible to move that into a `TryFromCtx` implementation on `ThunkKind`?

Reply:
The problem is that I need to get ord before getting name (which isn't part of ThunkKind) and then parse according to the ord value. Do you have an example of a similar situation ?

Reply:
Ah yes, you're right. I got the order wrong. Let's leave it like this then.